lib/pageUtils.js

```diff
@@ -0,0 +1,24 @@
+export const scrollTo = (element, to, duration) => {
+  if (duration <= 0) return
+  const difference = to - element.scrollTop;
+  const perTick = difference / duration * 16
+
+  window.requestAnimationFrame(() => {
+    element.scrollTop = element.scrollTop + perTick
+    if (element.scrollTop === to) return
+    scrollTo(element, to, duration - 16)
+  })
+}
+
+export const urlQuery = ((a) => {
+  if (a == '') return {}
+  let b = {}
+  for (let i = 0; i < a.length; ++i) {
+    let p=a[i].split('=', 2)
+    if (p.length == 1)
+      b[p[0]] = ''
+    else
+      b[p[0]] = decodeURIComponent(p[1].replace(/\+/g, ' '))
+  }
+  return b
```

Comment:
It's iterating over the length of the string, so it's clearer as a for loop. Otherwise I'd have to split the string into an array of characters (or do `Array.new(q.length).fill()`), reduce over that, but internally keep referring to the original string (because the internals need to operate on the whole string). Seems more convoluted to me (includes steps that don't do anything except set up the possibility of using `.reduce`.